src/coreclr/nativeaot/Runtime/unix/unixasmmacrosarm64.inc

```diff
@@ -226,6 +229,27 @@ C_FUNC(\Name):
     INLINE_GET_TLS_VAR \target, C_FUNC(tls_CurrentThread)
 .endm
 
+// Target cannot be x0.
+.macro INLINE_GET_ALLOC_CONTEXT target
+    .ifc \target, x0
+        .error "target cannot be x0"
+    .endif
```

Comment:
For emulated TLS we don't support target == x1 either. We should either fix the macro or update the check.